src/screens/login/components/ClouderyView.js

```diff
@@ -58,6 +65,25 @@ export const ClouderyView = ({
       }
     }
 
+    const isOidc = isOidcNavigationRequest(request)
+    if (isOidc) {
+      processOIDC(request)
+        .then(oidcResult => {
+          if (oidcResult.fqdn) {
+            startOidcOAuth(oidcResult.fqdn, oidcResult.code)
+          } else if (oidcResult.onboardUrl) {
+            startOidcOnboarding(oidcResult.onboardUrl, oidcResult.code)
+          }
+          return
+        })
+        .catch(error => {
+          if (error !== 'USER_CANCELED') {
+            setError(error.message, error)
+          }
+        })
+      return false
```

Comment:
I assume you don't await because `handleNavigation` can't be async, right? But can't we have a race condition later in the code?

Reply:
I don't think so.

First we return `false` so the navigation is aborted.

Then `startOidcOAuth` go to another step in LoginScreen, so this will unmount the ClouderyView component.
And `startOidcOnboarding` opens the InAppBrowser, so nothing should happen in the react-native side until the InAppBrowser is closed